portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,17 +15,61 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import java.util.ArrayList;
+
+import javax.servlet.ServletException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Entity;
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/comments")
 public class DataServlet extends HttpServlet {
+
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private Comment comment;
+
+  @Override
+  protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    resp.setContentType("application/json");
+    
+    Entity commentEntity = new Entity("Comment");
+
+    String name = null;
+    String commentText = null;
+    long timestamp = System.currentTimeMillis();
+    commentEntity.setProperty("name", name);
+    commentEntity.setProperty("comment", commentText);
+    commentEntity.setProperty("timestamp", timestamp);
+    commentEntity.setProperty("upvotes", 0);
+    commentEntity.setProperty("downvotes", 0);
+
+    datastore.put(commentEntity);
+
+    resp.sendRedirect("/index.html");
+  }
+
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Hisao!");
+    response.setContentType("application/html;");
+
+    ArrayList<Comment> recievedComments = new ArrayList<>();
+    Query query = new Query("Comments").addSort("timestamp", SortDirection.DESCENDING);
```

Comment:
Is this querying the right string? Based off `new Entity("Comment")` in the doPost function.

I would also recommend making "Comments" a hardcoded const, that helps out with having to maintain the hardcoded string in multiple places.

Reply:
I was actually thinking on hardcoding them as well, will do